include/irr/asset/material_compiler/CMaterialCompilerGLSLBackendCommon.h

```diff
@@ -0,0 +1,319 @@
+#ifndef __C_MITSUBA_MATERIAL_COMPILER_RAYTRACING_BACKEND_H_INCLUDED__
+#define __C_MITSUBA_MATERIAL_COMPILER_RAYTRACING_BACKEND_H_INCLUDED__
+
+#include <irr/asset/material_compiler/IR.h>
+#include <irr/asset/ICPUVirtualTexture.h>
+#include "os.h"
+
+namespace irr
+{
+namespace asset
+{
+namespace material_compiler
+{
+
+namespace instr_stream
```

Comment:
move to inside the backend class declaration, the streams will be different for each backend

Reply:
why would they be different??

Reply:
because a different backend may choose not to implement the same sort of state machine... i.e. VK raytracing/OptiX uses a Shader Binding Table and the software scheduler, so it might not be beneficial to use an instruction stream of the same granularity that we use for rasterized GLSL to avoid divergence.

Reply:
done?

Reply:
TODO

Reply:
whats the status on this?

Reply:
done